suitcase/fields.py

```diff
@@ -897,6 +912,14 @@ def unpack(self, data, **kwargs):
             self._value.append(structure)
             if data == b"":
                 break
+            if len(self._value) == self.num_elements:
+                break
+
+        if self.num_elements is not None and len(self._value) != self.num_elements:
+            raise SuitcaseParseError("Expected %r elements but received %r." %
```

Comment:
I'd recommend using %s and %d, respectively, in this format string. Using %r will add single quotes which could be confusing.